### ISSUE TITLE
[jak2] fortress rescue infinite loop crash

### ISSUE THREAD

**Issue by Zedb0T** [open-goal/jak-project#3151]
### Acknowledgements

- [X] I have [searched](https://github.com/open-goal/jak-project/issues?q=is%3Aissue+is%3Aopen+label%3Abug+label%3Ajak2+sort%3Aupdated-desc+) for duplicate or closed bug reports
- [X] I understand that I am supposed to provide my own legitimately obtained copy of the game

### Describe the Bug

Seen a few people get this now seems pretty rare but not super uncommon -> https://clips.twitch.tv/AlertMiniatureSrirachaAMPEnergyCherry-RGwZ0gVOucO9EIQK

### How To Reproduce

Play Jak 2 and get unlucky

### Does this problem occur on original hardware or PCSX2?

Yes, it's unique to OpenGOAL

### Expected Behavior

The game doesnt hang

### Environment Information

Observed on multiple environments.

### Game Version

NTSC 1.0 (black label)

### Have you set the game to something other than `60fps`?

No

**Comment by Zedb0T** [comment]
Ill keep updating this comment with video examples as I see them
https://clips.twitch.tv/AlertMiniatureSrirachaAMPEnergyCherry-RGwZ0gVOucO9EIQK

**Comment by ManDude** [comment]
This isn't a crash in the graphics thread, it's an infinite loop in goal.

**Comment by massimilianodelliubaldini** [comment]
I found a way to reproduce the issue somewhat readily using the debug menu.

1. Use the debug menu and hit `Game... > Continue... > fortress... > forrescb-mid`.
2. Exit the debug menu to teleport.
3. (Optional) Use the debug menu and hit `Actors... > Spawn Actors` and `Actors... > Reset Actors`. This disables all the guards in the room which makes it easier to move around, but it's not required to trigger the crash.
4. (Optional) Hit `Actors... > Regions Marks`. This allows you to see where `region-655` is but is also not required for the crash.
5. Use L2+R2 to hover Jak around freely:
    - Hug the wall with the blinking lights inside `region-655`.
    - Just run Jak against the wall for a bit, keeping the camera at about a 45 degree angle off the wall, and the infinite loop will eventually happen. These last 2 pieces are somehow the only way to trigger the loop.

I'm going to try a couple different approaches to see how specific or generic those steps need to be, maybe we can eliminate a few variables.

https://github.com/user-attachments/assets/42f71f52-6a2b-4c0f-989b-0e6659c15482

**Comment by trippjoe** [comment]
Discord user reported this issue after being shot by two krimzon guards simultaneously.

https://github.com/user-attachments/assets/183711a5-fc37-4977-9358-de15801ec064

**Comment by massimilianodelliubaldini** [comment]
I didn't try this, but if you despawn all the guards and try the "wall-running" movement, does the infinite loop still happen?

**Comment by massimilianodelliubaldini** [comment]
Yeah, can confirm it can still happen even when all actors - aside from Jak - are disabled.

**Comment by massimilianodelliubaldini** [comment]
Oh, game actually crashed this time, the infinite loop actually closed the exe. I don't think that's happened before. I'm on 0.2.27.

[jak2.2025-06-19T13-43-23.log](https://github.com/user-attachments/files/20823590/jak2.2025-06-19T13-43-23.log)
[jak2.2025-06-19T13-42-48.log](https://github.com/user-attachments/files/20823589/jak2.2025-06-19T13-42-48.log)

**Comment by massimilianodelliubaldini** [comment]
Found another way to reproduce. There are several walls with the same crashable property in this room.

1. Use the debug menu and hit `Game... > Continue... > fortress... > forrescb-mid`.
2. Exit the debug menu to teleport.
3. Use the debug menu and hit `Actors... > Spawn Actors` and `Actors... > Reset Actors`. This disables all the guards in the room which makes it easier to move around.
4. Go to the guards' platform at the end of this room, the triangular one flanked by consoles. Stand below the platform, hugging the wall.
5. Swivel the camera back and forth around Jak. Jak does not need to move, and the crash will still happen.
